### DISCUSSION TITLE
Can't find custom partial

### DISCUSSION THREAD

**Discussion by timmyd87** [josefarias/hotwire_combobox#116] [Q&A]
I'm now trying to render the results in a custom partial and looking at the docs (for async) we simply need to pass in 'render_in:' which looks like:

`<%= async_combobox_options @product_variants, render_in: { partial: "products/results" } %>`

however im getting an error

`ActionView::Template::Error (Missing partial products/_results with {:locale=>[:en], :formats=>[:turbo_stream], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :jbuilder]}.`

The partial definitely exists. 

![Screenshot 2024-03-21 at 1 07 34 pm](https://github.com/josefarias/hotwire_combobox/assets/12996091/d7411ffa-4ba0-4f6c-b19f-133df4f52ab8)

Any clues about where im going wrong?

**Comment by josefarias** [comment]
Hey @timmyd87 thanks for the question

Because the options are requested as a turbo_stream response, you need to either name your partial `_results.turbo_stream.erb` (note the html->turbo_stream change) or, if you're planning to use that partial in html responses, you can keep the current file name and pass an additional option to `:render_in` so it looks like this: `render_in: { partial: "products/results", formats: [:html] }`.

I'm fairly certain that's what's happening here. But if not, please share the whole error message. `ActionView::Template::Error` messages usually get an additional bit of context after a line break where it tells you where rails searched for the partial it couldn't find.

Closing for now but let me know!

**Reply by timmyd87** [reply]
That was it, thanks mate!

**Reply by Roxanoel** [reply]
I had the same issue today and it took me a while to figure it out, because in the docs it looks like the same partial file is used for both examples, there is no mention of needing to set the format or using a turbo_stream partial. Would it be possible to update the docs to reflect this? Happy to do it if it's not something you have time for at the moment!

**Reply by josefarias** [reply]
@Roxanoel ah yeah I see what you mean. Happy to accept a PR for that! Thank you!

I really need to make time to update the docs in general. Tons of missing features.

**Reply by Roxanoel** [reply]
Done, that should at least be a quick fix in the meantime 😊 
https://github.com/josefarias/hotwire_combobox_docs/pull/3
